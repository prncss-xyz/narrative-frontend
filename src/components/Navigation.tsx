import { useTheme } from "@emotion/react";
import { NavLink } from "react-router-dom";
import { Box, Flex } from "./basics";

export function Navigation() {
  const theme: any = useTheme();
  return (
    <Flex
      py={3}
      backgroundColor="tone5"
      color="tone3"
      zIndex={10}
      position="sticky"
      top="0px"
      gap={4}
      css={{
        "& .active": {
          fontWeight: "bold",
          // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-assignment
          color: theme.colors.tone0,
        },
      }}
    >
      <Box/>
      <Box>
        <NavLink to="/buy-order-list">Buy Orders</NavLink>
      </Box>
      <Box>
        <NavLink to="/dataset-list">Datasets</NavLink>
      </Box>
    </Flex>
  );
}
