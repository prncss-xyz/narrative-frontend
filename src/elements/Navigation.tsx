import { useTheme } from "@emotion/react";
import { NavLink } from "react-router-dom";
import { Flex, Box } from "./shared";

export default function Navigation() {
  const theme = useTheme() as any;
  return (
    <Flex
      py={3}
      backgroundColor="gray1"
      color="gray2"
      css={{
        zIndex: 10,
        position: "sticky",
        top: "0",
        "& .active": {
          fontWeight: "bold",
          color: theme.colors.white,
        },
      }}
    >
      <Box mx={4}>
        <NavLink to="/buy-orders">Buy Orders</NavLink>
      </Box>
      <Box mx={4}>
        <NavLink to="/datasets">Datasets</NavLink>
      </Box>
    </Flex>
  );
}
