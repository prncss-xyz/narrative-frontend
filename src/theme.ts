export default {
  colors: {
    gray1: "#707070",
    gray2: "#C0C0C0",
    gray25: "#D0D0D0",
    gray3: "#E0E0E0",
    black: "black",
    white: "white",
  },
  space: [0, 4, 8, 16, 32, 64, 128, 256, 512],
  sizes: [0, 32, 64, 128, 256, 512, 640],
  fontSizes: [12, 14, 16, 24, 32, 48, 64, 96, 128],
  radii: [0, 6, 12, 18],
};
