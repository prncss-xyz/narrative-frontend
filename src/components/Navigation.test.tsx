import { ThemeProvider } from "@emotion/react";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import theme from "../theme";
import { Navigation } from "./Navigation";

describe("Navigation", () => {
  it("should render correctly", () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <Router>
          <Navigation />
        </Router>
      </ThemeProvider>
    );
    expect(container).toMatchSnapshot();
  });
});
