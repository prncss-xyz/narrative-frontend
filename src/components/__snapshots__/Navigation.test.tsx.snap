// Vitest Snapshot v1

exports[`Navigation > should render correctly 1`] = `
<div>
  <div
    class="css-128lsy5-Box-Flex-Navigation e2i8k6n2"
    color="bgDark"
  >
    <div
      class="css-1x3sxzl-Box e2i8k6n3"
    />
    <div
      class="css-1x3sxzl-Box e2i8k6n3"
    >
      <a
        class=""
        href="/buy-order-list"
      >
        Buy Orders
      </a>
    </div>
    <div
      class="css-1x3sxzl-Box e2i8k6n3"
    >
      <a
        class=""
        href="/dataset-list"
      >
        Datasets
      </a>
    </div>
  </div>
</div>
`;
